### ISSUE TITLE
Add support for private fields (by not emitting them)

### ISSUE THREAD

**Issue by Julian** [numToStr/lemmy-help#13]
If a field is [marked private](https://emmylua.github.io/annotations/field.html) I think it'd be nice if they weren't emitted at all.

Right now it seems marking them as such is misparsed as the field name:

```
Pin                                                                        *Pin*


    Fields: ~
        {id}       (string)                @a label to identify the pin
        {private}  (__data_element)        Element
        {private}  (__element)             Element
        {private}  (__extmark)             number
        {private}  (__extmark_buf)         number
        {private}  (__extmark_hl_group)    string
        {private}  (__extmark_virt_text)   table
        {private}  (__ticker)              Ticker
        {private}  (__info)                Info
        {private}  (__ui_position_params)  UIParams
        {private}  (__use_widgets)         boolean
```

**Comment by Julian** [comment]
The `@` sign there I believe it seems is also misparsed, I presume it should just not appear in the output.

**Comment by numToStr** [comment]
> If a field is [marked private](https://emmylua.github.io/annotations/field.html) I think it'd be nice if they weren't emitted at all.

This is nice. I saw the `---@field` spec briefly but didn't implemented the scope thingy. Now I think this is useful and I can also use this. So I'm willing to implement it.

But one downside would be that you can't define a field named same as scope ie. `private`. XD

Also, I wonder how does lua-language-server renders that in the hover?

> The `@` sign there I believe it seems is also misparsed, I presume it should just not appear in the output.

Nope, that is expected. Everything after the `type` is treated as description. That means you can just remove the `@` char.

**Comment by Julian** [comment]
> Nope, that is expected. Everything after the type is treated as description. That means you can just remove the @ char.

But that's not what the linked page says, is it?

> ---@field [public|protected|private] field_name FIELD_TYPE[|OTHER_TYPE] [@comment]

The `@` sign is part of the syntax, it's how you declare the start of the comment portion, so it shouldn't appear in the output if I follow the page correctly.

**Comment by Julian** [comment]
> But one downside would be that you can't define a field named same as scope ie. private. XD

I don't think this is true FWIW, `@field private` would mean a field name of private which is public, since that's required syntax.

If you wanted a private field called private you have to write `@field private private` according to that line.

**Comment by numToStr** [comment]
> The @ sign is part of the syntax, it's how you declare the start of the comment portion, so it shouldn't appear in the output if I follow the page correctly.

hmmm....after seeing the example in the spec page, I guess you are right. But I'll say that it is better to remove it for now as it is not stripped, neither LLS is affected by it.

I'll have a look at the spec, and try to cover useful syntax in the future releases. So if you find anything missing which could be useful, please feel free to raise an issue

> If you wanted a private field called private you have to write @field private private according to that line.

Yes, They just need to adjust the comment :)
